tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -1807,19 +1809,14 @@ void StorageManager::sort_fragment_uris(std::vector<URI>* fragment_uris) const {
     assert(utils::parse::starts_with(fragment_name, "__"));
 
     // Get timestamp in the end of the name after '_'
-    assert(fragment_name.find_last_of("_") != std::string::npos);
+    assert(fragment_name.find_last_of('_') != std::string::npos);
     t_str = fragment_name.substr(fragment_name.find_last_of('_') + 1);
     sscanf(t_str.c_str(), "%lld", (long long int*)&t);
-    t_pos_vec.emplace_back(std::pair<uint64_t, uint64_t>(t, pos++));
+    sorted_fragment_uris->emplace_back(t, uri);
   }
 
   // Sort the names based on the timestamps
-  std::sort(t_pos_vec.begin(), t_pos_vec.end());
-  std::vector<URI> fragment_uris_sorted;
-  fragment_uris_sorted.resize(fragment_num);
-  for (uint64_t i = 0; i < fragment_num; ++i)
-    fragment_uris_sorted[i] = (*fragment_uris)[t_pos_vec[i].second];
-  *fragment_uris = fragment_uris_sorted;
+  std::sort(sorted_fragment_uris->begin(), sorted_fragment_uris->end());
```

Comment:
Why do we need to sort this list? It seems sufficient to just produce the list of parsed timestamp values in the output. Also, this is a lexicographic sort on the fragment name, which means it will sort first on the UUID and only then the timestamp.

Reply:
Note that the type of `sorted_fragment_uris` is `std::vector<std::pair<uint64_t, URI>>`. `std::sort` first attempts to sort on the first element of the pair (the timestamp) and then the second (the URI string, which essentially sorts lexicographically on UUID that appears first in the string after the URI parent that is common across fragments).